atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/TracingKeyValueService.java

```diff
@@ -0,0 +1,349 @@
+        try (CloseableTrace trace = startLocalTrace("getAllTableNames()")) {
+            return delegate().getAllTableNames();
+        }
+    }
+
+    @Override
+    public Multimap<Cell, Long> getAllTimestamps(TableReference tableRef, Set<Cell> keys, long timestamp) {
+        //noinspection unused - try-with-resources closes trace
+        try (CloseableTrace trace = startLocalTrace("getAllTimestamps({}, {} keys, ts {})",
+                tableRef, keys.size(), timestamp)) {
+            return delegate().getAllTimestamps(tableRef, keys, timestamp);
+        }
+    }
+
+    @Override
+    public Collection<? extends KeyValueService> getDelegates() {
+        //noinspection unused - try-with-resources closes trace
+        try (CloseableTrace trace = startLocalTrace("getDelegates()")) {
+            return ImmutableList.copyOf(Iterables.concat(ImmutableList.of(delegate()), delegate().getDelegates()));
+        }
+    }
+
+    @Override
+    public Map<RangeRequest, TokenBackedBasicResultsPage<RowResult<Value>, byte[]>> getFirstBatchForRanges(
+            TableReference tableRef,
+            Iterable<RangeRequest> rangeRequests,
+            long timestamp) {
+        //noinspection unused - try-with-resources closes trace
+        try (CloseableTrace trace = startLocalTrace("getFirstBatchForRanges({}, {} ranges, ts {})",
+                tableRef, Iterables.size(rangeRequests), timestamp)) {
+            return delegate().getFirstBatchForRanges(tableRef, rangeRequests, timestamp);
+        }
+    }
+
+    @Override
+    public Map<Cell, Long> getLatestTimestamps(TableReference tableRef,
+            Map<Cell, Long> timestampByCell) {
+        //noinspection unused - try-with-resources closes trace
+        try (CloseableTrace trace = startLocalTrace("getLatestTimestamps({}, {} cells)",
+                tableRef, timestampByCell.size())) {
+            return delegate().getLatestTimestamps(tableRef, timestampByCell);
+        }
+    }
+
+    @Override
+    public byte[] getMetadataForTable(TableReference tableRef) {
+        //noinspection unused - try-with-resources closes trace
+        try (CloseableTrace trace = startLocalTrace("getMetadataForTable({})", tableRef)) {
+            return delegate().getMetadataForTable(tableRef);
+        }
+    }
+
+    @Override
+    public Map<TableReference, byte[]> getMetadataForTables() {
+        //noinspection unused - try-with-resources closes trace
+        try (CloseableTrace trace = startLocalTrace("getMetadataForTables()")) {
+            return delegate().getMetadataForTables();
+        }
+    }
+
+    @Override
+    public ClosableIterator<RowResult<Value>> getRange(TableReference tableRef,
+            RangeRequest rangeRequest,
+            long timestamp) {
+        //noinspection unused - try-with-resources closes trace
+        try (CloseableTrace trace = startLocalTrace("getRange({}, {}, ts {})",
+                tableRef, rangeRequest, timestamp)) {
```

Comment:
So I followed this through the call hierarchy, and it looks like we lose the format-string when passing the string into `Tracer.startSpan`.

Whether we should redact depends on where the String resurfaces and how it is used.

If this eventually gets logged, and that log gets passed back via our log collection tool, then yes, we might want to redact. If the trace is only logged on the local machine, then I think we're OK to leave it there.